src/utils.ts

```diff
@@ -98,14 +101,92 @@ export const extractEvents = (
 
     switch (decodedEvent.name) {
       case InvariantEventNames.CreatePositionEvent:
-        eventsObject[InvariantEventNames.CreatePositionEvent].push(
-          parseEvent(decodedEvent)
-        );
+        const parsedCreateEvent: CreatePositionEvent = parseEvent(decodedEvent);
+        if (parsedCreateEvent.pool.toString() !== PROMOTED_POOL.toString()) {
+          return;
+        }
+        if (!!eventsObject[parsedCreateEvent.owner.toString()]) {
+          const correspondingItem = eventsObject[
+            parsedCreateEvent.owner.toString()
+          ].closed.find(
+            (item) =>
+              item.events[1].id.toString() === parsedCreateEvent.id.toString()
+          );
+          if (correspondingItem) {
+            const correspondingIndex = eventsObject[
+              parsedCreateEvent.owner.toString()
+            ].closed.findIndex(
+              (item) =>
+                item.events[1].id.toString() === parsedCreateEvent.id.toString()
+            );
+            eventsObject[parsedCreateEvent.owner.toString()].closed.splice(
+              correspondingIndex,
+              1
+            );
+            eventsObject[parsedCreateEvent.owner.toString()].closed.push({
+              events: [parsedCreateEvent, correspondingItem.events[1]],
+            });
+            return;
+          }
+          eventsObject[parsedCreateEvent.owner.toString()].active.push({
+            event: parsedCreateEvent,
+          });
+          return;
+        }
+        eventsObject[parsedCreateEvent.owner.toString()] = {
+          active: [
+            {
+              event: parsedCreateEvent,
+            },
+          ],
+          closed: [],
+        };
         break;
       case InvariantEventNames.RemovePositionEvent:
-        eventsObject[InvariantEventNames.RemovePositionEvent].push(
-          parseEvent(decodedEvent)
-        );
+        //@ts-expect-error
+        const parsedRemoveEvent: RemovePositionEvent = parseEvent(decodedEvent);
```

Comment:
It will be better to don't use the expect error, you should be able to cast it with `as RemovePositionEvent` w/o any typescript warnings